torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1669,7 +1672,8 @@ std::vector<kir::Val*> Index::getNonGlobalProducerStridedIndices(
   for (auto root_id : root_dom) {
     // Already taken care of because we can detect no indexing required
     if (root_id->isBroadcast() || root_id->isReduction() ||
-        gpu_lower->trivialReductionInfo().isDerived(root_id)) {
+        gpu_lower->trivialReductionInfo().isDerived(root_id) ||
+        root_id->isStride()) {
```

Comment:
So trivial reductions should be marked as a stride iter domain as a cleanup PR? ;-)

Reply:
Yeah, some cleanup would be nice. We frequently do `isBroadcast || isReduction || isTrivialReduction || isStride`. `TensorDomain::noReductions` helps but not enough.